apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -178,111 +151,94 @@ export class UpsertWorkflowUseCase {
     };
   }
 
-  private async mapSteps(
-    workflowOrigin: WorkflowOriginEnum,
-    user: UserSessionData,
-    commandWorkflowSteps: Array<StepCreateDto | StepUpdateDto>,
-    persistedWorkflow?: NotificationTemplateEntity | undefined
+  private async buildSteps(
+    command: UpsertWorkflowCommand,
```

Comment:
Notice that too avoid multiple arguments for each private function and ad-hoc types, the command object is passe to all of them so that each function can pick whatever it needs from it.